src/ssh/azext_ssh/custom.py

```diff
@@ -0,0 +1,115 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import functools
+import os
+import hashlib
+import json
+
+from knack import util
+
+from . import ip_utils
+from . import rsa_parser
+from . import ssh_utils
+
+
+def ssh_vm(cmd, resource_group_name=None, vm_name=None, ssh_ip=None, public_key_file=None, private_key_file=None):
+    _do_ssh_op(cmd, resource_group_name, vm_name, ssh_ip,
+               public_key_file, private_key_file, ssh_utils.start_ssh_connection)
+
+
+def ssh_config(cmd, config_path, resource_group_name=None, vm_name=None, ssh_ip=None,
+               public_key_file=None, private_key_file=None):
+    op_call = functools.partial(ssh_utils.write_ssh_config, config_path, resource_group_name, vm_name)
+    _do_ssh_op(cmd, resource_group_name, vm_name, ssh_ip, public_key_file, private_key_file, op_call)
+
+
+def _do_ssh_op(cmd, resource_group, vm_name, ssh_ip, public_key_file, private_key_file, op_call):
+    _assert_args(resource_group, vm_name, ssh_ip)
+    public_key_file, private_key_file = _check_public_private_files(public_key_file, private_key_file)
+    ssh_ip = ssh_ip or ip_utils.get_ssh_ip(cmd, resource_group, vm_name)
+
+    if not ssh_ip:
+        raise util.CLIError(f"VM '{vm_name}' does not have a public IP address to SSH to")
+
+    scopes = ["https://pas.windows.net/CheckMyAccess/Linux/user_impersonation"]
+    data = _prepare_jwk_data(public_key_file)
+    from azure.cli.core._profile import Profile
+    profile = Profile(cli_ctx=cmd.cli_ctx)
+    username, certificate = profile.get_msal_token(scopes, data)
+
+    cert_file = _write_cert_file(public_key_file, username, certificate)
```

Comment:
Hi @rlrossiter , previously there is a fixed `ssh-rsa-cert-v01@openssh.com ` in certificate head, which is weird to me. I replace it with `username`, could you please help check whether it is right ?

**Previous code in azure-cli-core**
```
class SSHCredentials(object):
    def __init__(self, username, cert):
        self.username = username
        self.certificate = "ssh-rsa-cert-v01@openssh.com " + cert
```

Reply:
The original was correct as that was the type of certificate the file was (it's required by SSH). The username is encoded as part of the base64 bytes written to the file. The username was passed back alongside the credentials but not part of the certificate in order to know who to SSH as

Reply:
Thanks for your explain. Yes, I decode the certificate and find `ssh-ras-cert-v01@openssh.com` is in it. I have change it back. You can have a look.

Could you please share more about this? Is there any document which explain it? It's still weird to me that why it is a fixed name in the certificate. Thanks in advance.